tiledb/sm/filesystem/azure.cc

```diff
@@ -165,39 +173,133 @@ Status Azure::empty_container(const URI& container) const {
 Status Azure::flush_blob(const URI& uri) {
   assert(client_);
 
+  if (!use_block_list_upload_) {
+    return flush_blob_direct(uri);
+  }
+
+  if (!uri.is_azure()) {
+    return LOG_STATUS(Status::AzureError(
+        std::string("URI is not an Azure URI: " + uri.to_string())));
+  }
+
+  Buffer* const write_cache_buffer = get_write_cache_buffer(uri.to_string());
+
+  const Status flush_write_cache_st =
+      flush_write_cache(uri, write_cache_buffer, true);
+
+  // We do not need to protect 'block_list_upload_states_' here because
+  // 'count' is thread-safe.
+  if (block_list_upload_states_.count(uri.to_string()) == 0) {
+    return flush_write_cache_st;
+  }
+
+  std::string container_name;
+  std::string blob_path;
+  RETURN_NOT_OK(parse_azure_uri(uri, &container_name, &blob_path));
+
+  // We do not need to protect 'block_list_upload_states_' here because
+  // 'at' is thread-safe. This class only guarantees thread-safety among
```

Comment:
I'll update the comment to indicate that we do not use the 'block_list_upload_states_lock_' to make it clearer that we would otherwise be using a lock.